### DISCUSSION TITLE
Design patterns and software design discussion

### DISCUSSION THREAD

**Discussion by jaxernst** [jaxernst/Blockchain-Boids#11] [Ideas]
How can the architecture be improved on?

**Comment by jaxernst** [comment]
I don't know js well so give some feedback here. @hunterbryant

**Comment by hunterbryant** [comment]
Check out this writeup here:
[3D Boids](https://ercangercek.medium.com/experiment-with-3d-boids-and-javascript-fe8fa51707b8)

**Reply by hunterbryant** [reply]
I think a lot of the performance gains you need come down to efficient spatial partitioning

**Reply by jaxernst** [reply]
Thats actual insanity

**Reply by jaxernst** [reply]
I did spatial partitioning for my python implementation but it can be tricky to do it right. Once the basic boid rules are done we should look more into optimizing
